### PR TITLE
Obsolete AssemblyName.CodeBase, AssemblyName.EscapedCodeBase

### LINKED ISSUE #42857: Assembly.CodeBase obsoletion is incomplete
`Assembly.CodeBase` and `Assembly.EscapedCodeBase` were marked obsolete in https://github.com/dotnet/runtime/issues/31127

but this work was incomplete: `AssemblyName.CodeBase` and `AssemblyName.EscapedCodeBase` are not marked. In addition, usages of these obsolete APIs remain in the source code.

We need to carefully audit all the use-sites and either remove them or provide explicit justification for our continued use.

### PR DESCRIPTION
**Pull request by buyaa-n**
AssemblyName.CodeBase is related to Assembly.CodeBase which is obsoleted in 7.0, same for AssemblyName.EscapedCodeBase

Loading assembly using this property is not supported in .Net Core which is main functional usage of this API

```cs
private static Assembly LoadAssembly(string codeBase)
{
    AssemblyName assemblyName = new AssemblyName { CodeBase = assemblyPath };
    return Assembly.Load(assemblyName);
}
```
Fixes https://github.com/dotnet/runtime/issues/42857

### CODE REVIEW

**Review by agocke — APPROVED** [review]
LGTM

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs [hunk 1]
```diff
@@ -134,9 +134,9 @@ public override AssemblyName GetName(bool copiedName)
 
             an.RawFlags = GetFlags() | AssemblyNameFlags.PublicKey;
 
-#pragma warning disable IL3000 // System.Reflection.AssemblyName.CodeBase' always returns an empty string for assemblies embedded in a single-file app.
+#pragma warning disable IL3000, SYSLIB0044 // System.Reflection.AssemblyName.CodeBase' always returns an empty string for assemblies embedded in a single-file app.
             an.CodeBase = GetCodeBase();
-#pragma warning restore IL3000
+#pragma warning restore IL3000, SYSLIB004 // AssemblyName.CodeBase and AssemblyName.EscapedCodeBase are obsolete, using them for loading an assembly is not supported.
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs:139]: This comment should be on the `pragma warning disable`

File: src/libraries/Common/src/System/Obsoletions.cs [hunk 2]
```diff
@@ -141,5 +141,8 @@ internal static class Obsoletions
 
         internal const string EcDhPublicKeyBlobMessage = "ECDiffieHellmanPublicKey.ToByteArray() and the associated constructor do not have a consistent and interoperable implementation on all platforms. Use ECDiffieHellmanPublicKey.ExportSubjectPublicKeyInfo() instead.";
         internal const string EcDhPublicKeyBlobDiagId = "SYSLIB0043";
+
+        internal const string AssemblyNameCodeBaseMessage = "AssemblyName.CodeBase and AssemblyName.EscapedCodeBase are obsolete, using them for loading an assembly is not supported.";
```
[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Common/src/System/Obsoletions.cs:145]: Probably could say try produce `location` from `codeBase` (like try create a `Uri`  from `codeBase` and get the `LocalPath`) and use the API that supports location for getting `Assembly`
```cs
string path = assemblyName.CodeBase;
try
{
    path = new Uri(codeBase).LocalPath;
}
catch { }
Assembly assembly = Assembly.LoadFrom(path);
```

[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Common/src/System/Obsoletions.cs]: @stephentoub would you suggest a different message?

[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Common/src/System/Obsoletions.cs]: > Is there something we can say about what should be used / done instead

I think the problem for suggesting a workaround is there is no one workaround that could work all the time, it depend if the `AssemblyName.CodeBase` has a Uri path or any other path structure. Therefore I could not come up with a better suggestion, unless you guys suggest something.

[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/Common/src/System/Obsoletions.cs]: Ok

**Comment by buyaa-n** [comment]
The failure is unrelated and [reported](https://github.com/dotnet/runtime/issues/70692), merging

**Comment by gewarren** [comment]
> > Does this need a breaking change issue?
> 
> I don't think so, no change in made in existing implementation. Was the Assembly.CodeBase, Assembly.EscapedCodeBase obsoletion caused a breaking change?

SYSLIB0012 was documented as a breaking change [here](https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/obsolete-apis-with-custom-diagnostics). I will do the same thing for the .NET 7 page for SYSLIB0044.

**Comment by buyaa-n** [comment]
I see, it's for obsoleted warnings, got it

**Pull request merged**